app/src/pages/Lobby/index.tsx

```diff
@@ -41,181 +32,29 @@ const useStyles = makeStyles((theme: Theme) =>
-                value={String(currentGame.num_entries_per_player || "")}
-              />
-            </Grid>
-            <Grid item>
-              <LetterInput value={currentGame.starting_letter || ""} />
-            </Grid>
-            <Grid item>
-              <ScreenCardsCheckbox value={Boolean(currentGame.screen_cards)} />
-            </Grid>
-
-            <Grid item>
-              <Typography variant="h6" className={titleClasses.title}>
-                Turns
-              </Typography>
-            </Grid>
-            <Grid item>
-              <SecondsPerTurnInput
-                value={String(currentGame.seconds_per_turn || "")}
-              />
-            </Grid>
-            <Grid item>
-              <AllowCardSkipsCheckbox
-                value={Boolean(currentGame.allow_card_skips)}
-              />
-            </Grid>
-            <Grid item>
-              <Typography variant="h6" className={titleClasses.title}>
-                Rounds
-              </Typography>
-              <Box pl={2} pt={1} fontSize="0.75rem" color={grey[600]}>
-                {currentPlayer.role === PlayerRole.Host &&
-                  "You can add, remove, or reorder rounds. By default, cards submitted will be re-used across rounds of Taboo, Charades, and Password."}
-              </Box>
-            </Grid>
-            <Grid item>
-              {currentPlayer.role === PlayerRole.Host ? (
-                <ControllableRoundSettings />
-              ) : (
-                <RoundSettings />
-              )}
-            </Grid>
-            <Grid item />
-          </Grid>
-        </Grid>
-      </ExpansionPanelDetails>
-    </ExpansionPanel>
-  )
-}
-
-function WaitingRoomSection() {
-  const currentPlayer = React.useContext(CurrentPlayerContext)
-  const titleClasses = useTitleStyle()
-
-  return (
-    <Grid item>
-      <Grid container spacing={2} direction="column">
-        <Grid item>
-          <Typography variant="h4" className={titleClasses.title}>
-            Lobby
-          </Typography>
-        </Grid>
-        <Grid item>
-          <UsernameInput
-            username={currentPlayer.username || ""}
-            playerId={currentPlayer.id}
-          />
-        </Grid>
-        <WaitingRoom></WaitingRoom>
-      </Grid>
-    </Grid>
-  )
-}
-
 function Lobby() {
   const currentPlayer = React.useContext(CurrentPlayerContext)
+  const currentGame = React.useContext(CurrentGameContext)
   const [updatePlayer] = useUpdatePlayerMutation()
   const classes = useStyles()
   const titleClasses = useTitleStyle()
   const location = useLocation()
   const [hideShare, setHideShare] = React.useState(false)
+  const [updateGameSettings] = useUpdateGameSettingsMutation()
+  const [cardPlayStyle, setCardPlayStyle] = React.useState(
+    GameCardPlayStyleEnum.PlayersSubmitWords
+  )
+
+  React.useEffect(() => {
+    setCardPlayStyle(currentGame.card_play_style)
+  }, [currentGame.card_play_style])
```

Comment:
need to sort out updating the cache, vs. haivng local state -- fix this anti pattern in a number of places around Settings code

Reply:
Yeah, you might benefit from wrapping the [`games` setting properties](https://github.com/avimoondra/fishbowl/blob/6d57c3f7a7820fbf0eeaa475a068563d5d71d976/app/src/pages/Lobby/operations.graphql#L3-L10) in a fragment. You could pass that around to components (or throw it in the `CurrentGameContext`?) for use in optimistic responses.

Reply:
Yea definitely - this was the route I was heading directionally - https://github.com/avimoondra/fishbowl/pull/122, lmk what you think (sans fragments yet)